pandas/io/json/json.py

```diff
@@ -332,14 +357,22 @@ def read_json(path_or_buf=None, orient=None, typ='frame', dtype=True,
             exists = False
 
         if exists:
-            fh, handles = _get_handle(filepath_or_buffer, 'r',
-                                      encoding=encoding)
```

Comment:
So I would change this around a bit. Make this the JsonReader. and pass it both lines and chunksize (as well as all the other params).

So you *always* create it.

Then if you have lines & chunksize you can return an iterator, otherwise you can simply return it like this

```
jr = JsonReader(.....)


if lines & chunksize:
     return jr

return jr.read()
```

Then you can make ``_get_obj`` a method on JsonReader. Much cleaner this way.

This is much more like how ``pandas.io.pytables.TableIterator`` (and read_csv iterator) are setup. You *always* create a reader, and either return it or just call .read() on it.